core/modules/user/store/actions.ts

```diff
@@ -257,12 +258,20 @@ const actions: ActionTree<UserState, RootState> = {
       return ordersHistory
     }
   },
-  async refreshOrdersHistory ({ commit }, { resolvedFromCache, pageSize = 20, currentPage = 1 }) {
+  async refreshOrdersHistory ({ commit, getters }, { resolvedFromCache, pageSize = 20, currentPage = 1 }) {
     const resp = await UserService.getOrdersHistory(pageSize, currentPage)
 
     if (resp.code === 200) {
-      commit(types.USER_ORDERS_HISTORY_LOADED, resp.result) // this also stores the current user to localForage
-      EventBus.$emit('user-after-loaded-orders', resp.result)
+      let oldOrders = getters.getOrdersHistory;
+      let orders = resp.result;
+      if (oldOrders && orders.items) {
+        orders.items = [
+          ...oldOrders,
+          ...resp.result.items
+        ]
+      }
+      commit(types.USER_ORDERS_HISTORY_LOADED, orders) // this also stores the current user to localForage
+      EventBus.$emit('user-after-loaded-orders', orders)
     }
 
     if (!resolvedFromCache) {
```

Comment:
Please add return before this `Promise.resolve`

Reply:
modified it to return Promise.resolve....